t/instant_answer_basics.t

```diff
@@ -0,0 +1,165 @@
+        return decode_json( $ia_update_request->decoded_content )->{result};
+    };
+
+    # Create user
+    my $user_request = $cb->(
+        POST '/testutils/new_user',
+        { username => 'ia_user' }
+    );
+    ok( $user_request->is_success, 'Creating an IA user' );
+
+    # Get session
+    my $session_request = $cb->(
+        POST '/testutils/user_session',
+        { username => 'ia_user' }
+    );
+    ok( $session_request->is_success, 'Getting IA user Cookie' );
+    my $cookie = 'ddgc_session=' . $session_request->content;
+
+    # Create an IA
+    my $new_ia_req = $cb->(
+        POST '/ia/create',
+        Cookie          => $cookie,
+        Content         => [ data => encode_json( {
+            id            => 'test_ia',
+            name          => 'Test IA',
+            description   => 'This is a test IA',
+            example_query => 'testing IAs',
+            repo          => 'longtail',
+            action_token  => $get_action_token->( $cookie ),
+        } ) ],
+    );
+    ok( $new_ia_req->is_success, 'Creating IA' );
+
+    # Find IA
+    my $ia = $d->rs('InstantAnswer')->find('test_ia');
+    ok( $ia, 'Query returns something' );
+    isa_ok( $ia, 'DDGC::DB::Result::InstantAnswer' );
+
+    # Check all_milestones JSON (should have everything)
+    my $ia_repo = $get_repo_json->({ repo => 'longtail', all_milestones => 1 });
+    is( ref $ia_repo->{test_ia}, 'HASH', 'Test IA is in all milestone repo JSON' );
+
+    # "Approve" IA
+    my $ia_approve_result = $set_ia_value->({ cookie => $cookie, public => 1 });
+    is( $ia_approve_result->{saved}, 1, 'IA "Approve" was successful' );
+
+    # This forces a re-select from the db - see: https://metacpan.org/pod/DBIx::Class::Row#discard_changes
+    $ia->discard_changes;
+    is( $ia->public, 1, 'IA Result is now public' );
+
+    # Check repo JSON - test_ia should be absent due to its milestone
+    $ia_repo = $get_repo_json->({ repo => 'longtail' });
+    ok( !$ia_repo->{test_ia}, 'Test IA not in published milestone' );
+
+    # Set IA milestone
+    my $ia_milestone_result = $set_ia_value->({ cookie => $cookie, dev_milestone => 'testing' });
+    is( $ia_milestone_result->{saved}, 1, 'Set test_ia milestone to testing' );
+
+    # Check repo JSON - test_ia should now be present
+    $ia_repo = $get_repo_json->({ repo => 'longtail' });
+    is( ref $ia_repo->{test_ia}, 'HASH', 'Test IA is now in published milestone' );
+
+    # Use this when you need to see session data
+    # $cb->( GET '/testutils/debug_session' );
+
+};
+
+# Some basic backend template checks
+my $mech = Test::WWW::Mechanize::PSGI->new( app => $app );
+
+my $ia = $d->rs('InstantAnswer')->find('test_ia');
+
+$ia->update({ dev_milestone => 'live', perl_module => 'DDG::Longtail::TestIA' });
+$ia->add_to_topics( { name => 'interesting_topic' } );
+
+$mech->get_ok('/ia');
+$mech->content_contains('test_ia');
+
+$mech->get_ok('/ia/view/test_ia');
+$mech->title_is('Test Ia'); # Title case filter
```

Comment:
This is an alternative approach to [this page content check in blog.t](https://github.com/duckduckgo/community-platform/blob/cc7a22b/t/blog.t#L90-L93).

[Test::WWW::Mechanize::PSGI](https://metacpan.org/pod/Test::WWW::Mechanize::PSGI) can completely replace or augment Plack::Test depending on your preference or which types of tests you're going to perform.